modules/sdk-core/src/bitgo/pendingApproval/pendingApproval.ts

```diff
@@ -374,4 +300,136 @@ export class PendingApproval implements IPendingApproval {
+      return false;
+    }
+
+    // If there are no recipients, then the transaction cannot be recreated
+    const recipients = this.info()?.transactionRequest?.buildParams?.recipients || [];
+    const type = this.info()?.transactionRequest?.buildParams?.type;
+
+    // We only want to not recreate transactions with no recipients if it is a UTXO coin.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return !(
+      utxolib.isValidNetwork((this.baseCoin as any).network) &&
+      recipients.length === 0 &&
+      type !== 'consolidate'
+    );
+  }
+
+  /*
+   * Internal helper function to get the serialized transaction which is being approved.
+   * If this PA is of type 'transactionRequest' this function will try to rebuild and resign the transaction
+   * @param {ApproveOptions} params
+   * @param {boolean} canRecreateTransaction -
+   * @param {RequestTracer} reqId id tracer
+   */
+  private async preApprove(params: ApproveOptions = {}, reqId: IRequestTracer): Promise<PreApproveResult | undefined> {
+    // TransactionRequestLite or Multisig tx's must sign before pending approval is approved
+    // Re-signed tx is provided to the pending approval api
+    if (this.type() === Type.TRANSACTION_REQUEST) {
+      /*
+       * If this is a request for approving a transaction, depending on whether this user has a private key to the wallet
+       * (some admins may not have the spend permission), the transaction could either be rebroadcast as is, or it could
+       * be reconstructed. It is preferable to reconstruct a tx in order to adhere to the latest network conditions
+       * such as newer unspents, different fees, or a higher sequence id
+       */
+      if (params.tx) {
+        // the approval tx was reconstructed and explicitly specified - pass it through
+        return {
+          txHex: params.tx,
+        };
+      }
+
+      const transaction = _.get(
+        this.info(),
+        `transactionRequest.coinSpecific.${this.baseCoin.type}`
+      ) as PreApproveResult;
+
+      // this user may not have spending privileges or a passphrase may not have been passed in
+      if (!this.canRecreateTransaction(params)) {
+        if (!_.isObject(transaction)) {
+          throw new Error('there is neither an original transaction object nor can a new one be recreated');
+        }
+        return transaction;
+      }
+
+      if (this._pendingApproval.txRequestId) {
+        return await this.recreateAndSignTSSTransaction(params, reqId);
+      }
+      return await this.recreateAndSignTransaction(params);
+    }
+  }
+
+  /**
+   * Internal helper function to perform any post-approval actions.
+   * If type is 'transactionRequestFull', this will sign the txRequestFull if possible
+   * @param params
+   * @param reqId
+   * @private
+   */
+  private async postApprove(params: ApproveOptions = {}, reqId: IRequestTracer): Promise<void> {
```

Comment:
this is fine, but i did get a little confused when i was reading the function that uses it, because this reads like we are using a post method to approve

Reply:
we can re-name the pre and post to something like `handlePreApproval` and `handlePostApproval` ?

Reply:
i think that reads a bit more clear :)